### PR TITLE
Change computational coordinates naming convention

### PR DESCRIPTION
**Pull request by jkozdon**
The coordinate system used for the computation is now: `(x1, x2, x3)`

The element reference system now: `(ξ1, ξ2, ξ3)`

Note: That the old drivers in `test/DGmethods_old` have not been updated to new naming convention

### CODE REVIEW

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
Would be great to have another set of eyes look this over (cc @lcw @tapios)

**Comment by tapios** [comment]
Looks good to me. Thanks so much for doing this so promptly! (Would you mind also updating the notation in section 7 of the design doc on Overleaf?)

File: src/Atmos/Model/AtmosModel.jl [hunk 1]
```diff
@@ -63,7 +63,7 @@ function vars_diffusive(m::AtmosModel, T)
 end
 function vars_aux(m::AtmosModel, T)
   @vars begin
-    coord::@vars(x::T,y::T,z::T)
+    coord::@vars(x1::T,x2::T,x3::T)
```
Comment (simonbyrne) [inline src/Atmos/Model/AtmosModel.jl]: might be simpler to just make this an `SVector`

File: src/Atmos/Model/moisture.jl [hunk 2]
```diff
@@ -22,7 +22,7 @@ function internal_energy(m::MoistureModel, state::Vars, aux::Vars)
   q_pt = get_phase_partition(m, state)
   ρinv = 1 / state.ρ
   ρe_kin = ρinv*sum(abs2, state.ρu)/2
-  ρe_pot = state.ρ * grav * aux.coord.z
+  ρe_pot = state.ρ * grav * aux.coord[3]
```
Comment (lcw) [inline src/Atmos/Model/moisture.jl]: Note that `aux.coord[3]` is not the radial direction.  This code needs to change to work on the sphere.

Comment (simonbyrne) [reply src/Atmos/Model/moisture.jl]: Yes: we need a way that will work for both the rectangle and sphere

Comment (jkozdon) [reply src/Atmos/Model/moisture.jl]: My recommendation would be that we store `ϕ` and `∇ϕ`. We can then check in one place how to set `ϕ` (probably in `init_aux` if the grid type is sphere)

File: src/Atmos/Model/AtmosModel.jl [hunk 3]
```diff
@@ -63,7 +63,7 @@ function vars_diffusive(m::AtmosModel, T)
 end
 function vars_aux(m::AtmosModel, T)
   @vars begin
-    coord::@vars(x::T,y::T,z::T)
+    coord::::SVector{3,T}
```
Comment (jkozdon) [inline src/Atmos/Model/AtmosModel.jl]:
```suggestion
    coord::SVector{3,T}
```

Comment (simonbyrne) [reply src/Atmos/Model/AtmosModel.jl]: Thanks, i should make hasty PRs while running out the door.

File: src/Atmos/Model/moisture.jl (hunk 2 above)
Comment (simonbyrne) [reply src/Atmos/Model/moisture.jl]: Agreed, that does seem like the simplest solution for now.

**Comment by jkozdon** [comment]
Should we merge this?

**Comment by charleskawczynski** [comment]
I'm reading over it now

**Comment by jkozdon** [comment]
> Would you mind also updating the notation in section 7 of the design doc on Overleaf?

Done.

File: test/Mesh/Metrics.jl [hunk 4]
```diff
@@ -676,67 +676,67 @@ end
...
-       yr, ys, yt,
-       zr, zs, zt) = (fxr(x,y,z), fxs(x,y,z), fxt(x,y,z),
-                      fyr(x,y,z), fys(x,y,z), fyt(x,y,z),
-                      fzr(x,y,z), fzs(x,y,z), fzt(x,y,z))
-      J = (xr .* (ys .* zt - yt .* zs) +
-           yr .* (zs .* xt - zt .* xs) +
-           zr .* (xs .* yt - xt .* ys))
+      (x1ξ1, x1ξ2, x1ξ3,
+       x2ξ1, x2ξ2, x2ξ3,
+       x3ξ1, x3ξ2, x3ξ3) = (fx1ξ1(x1,x2,x3), fx1ξ2(x1,x2,x3), fx1ξ3(x1,x2,x3),
```
Comment (charleskawczynski) [inline test/Mesh/Metrics.jl]: Maybe align these?
```
       x3ξ1, x3ξ2, x3ξ3) = (fx1ξ1(x1,x2,x3), fx1ξ2(x1,x2,x3), fx1ξ3(x1,x2,x3),
                            fx2ξ1(x1,x2,x3), fx2ξ2(x1,x2,x3), fx2ξ3(x1,x2,x3),
                            fx3ξ1(x1,x2,x3), fx3ξ2(x1,x2,x3), fx3ξ3(x1,x2,x3))
```

File: test/Mesh/Metrics.jl [hunk 5]
```diff
@@ -676,67 +676,67 @@ end
...
-      @test sJ[ind,4,:] .* ny[ind,4,:] ≈  J[:,Nq,:,:] .* ηy[:,Nq,:,:]
-      @test sJ[ind,4,:] .* nz[ind,4,:] ≈  J[:,Nq,:,:] .* ηz[:,Nq,:,:]
-      @test sJ[ind,5,:] .* nx[ind,5,:] ≈ -J[:,:, 1,:] .* ζx[:,:, 1,:]
-      @test sJ[ind,5,:] .* ny[ind,5,:] ≈ -J[:,:, 1,:] .* ζy[:,:, 1,:]
-      @test sJ[ind,5,:] .* nz[ind,5,:] ≈ -J[:,:, 1,:] .* ζz[:,:, 1,:]
-      @test sJ[ind,6,:] .* nx[ind,6,:] ≈  J[:,:,Nq,:] .* ζx[:,:,Nq,:]
-      @test sJ[ind,6,:] .* ny[ind,6,:] ≈  J[:,:,Nq,:] .* ζy[:,:,Nq,:]
-      @test sJ[ind,6,:] .* nz[ind,6,:] ≈  J[:,:,Nq,:] .* ζz[:,:,Nq,:]
+      @test hypot.(n1, n2, n3) ≈ ones(T, size(n1))
+      @test ([sJ[ind,1,:] .* n1[ind,1,:], sJ[ind,1,:] .* n2[ind,1,:],
```
Comment (charleskawczynski) [inline test/Mesh/Metrics.jl]: Could align these also:
```      @test ([sJ[ind,1,:] .* n1[ind,1,:],
              sJ[ind,1,:] .* n2[ind,1,:],
              sJ[ind,1,:] .* n3[ind,1,:]] ≈
             [-J[ 1,:,:,:] .* ξ1x1[ 1,:,:,:],
              -J[ 1,:,:,:] .* ξ1x2[ 1,:,:,:],
              -J[ 1,:,:,:] .* ξ1x3[ 1,:,:,:]])
      @test ([sJ[ind,2,:] .* n1[ind,2,:],
              sJ[ind,2,:] .* n2[ind,2,:],
              sJ[ind,2,:] .* n3[ind,2,:]] ≈
             [ J[Nq,:,:,:] .* ξ1x1[Nq,:,:,:],
               J[Nq,:,:,:] .* ξ1x2[Nq,:,:,:],
               J[Nq,:,:,:] .* ξ1x3[Nq,:,:,:]])
```

**Comment by jkozdon** [comment]
> Alright, I looked through it and left some reviews.

@charleskawczynski Thanks for the careful read through. I've made your changes.

**Pull request merged**